core/server/inventory_test.go

```diff
@@ -317,3 +320,188 @@ func TestGetInventoryHelmReleaseWithKubeconfig(t *testing.T) {
+	// inv lives at status.inventory.entries
+	stdErr := errors.New("no status.inventory found on resource, it hasn't been synced yet or is not queryable from this endpoint")
+
+	testCases := []struct {
+		name        string
+		obj         *unstructured.Unstructured
+		expected    []*unstructured.Unstructured
+		expectedErr error
+	}{
+		{
+			name:        "no status field",
+			obj:         &unstructured.Unstructured{},
+			expected:    nil,
+			expectedErr: stdErr,
+		},
+		{
+			name: "empty status",
+			obj: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"status": map[string]interface{}{},
+				},
+			},
+			expected:    nil,
+			expectedErr: stdErr,
+		},
+		{
+			name: "empty inventory",
+			obj: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"status": map[string]interface{}{
+						"inventory": map[string]interface{}{},
+					},
+				},
+			},
+			expected: nil,
+		},
+		{
+			name: "empty entry item",
+			obj: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"status": map[string]interface{}{
+						"inventory": map[string]interface{}{
+							"entries": []interface{}{
+								map[string]interface{}{},
+							},
+						},
+					},
+				},
+			},
+			expected:    nil,
+			expectedErr: errors.New("unable to parse stored object metadata: "),
+		},
+		{
+			name: "invalid inventory",
+			obj: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"status": map[string]interface{}{
+						"inventory": map[string]interface{}{
+							"entries": []interface{}{
+								map[string]interface{}{
+									"v":  "v1",
+									"id": "foo",
+								},
+							},
+						},
+					},
+				},
+			},
+			expected:    nil,
+			expectedErr: errors.New("unable to parse stored object metadata: foo"),
+		},
+		{
+			name: "valid inventory",
+			obj: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"status": map[string]interface{}{
+						"inventory": map[string]interface{}{
+							"entries": []interface{}{
+								map[string]interface{}{
+									"v":  "v1",
+									"id": "my-namespace_my-deployment_apps_Deployment",
+								},
+								map[string]interface{}{
+									"v":  "v1",
+									"id": "my-other-namespace_my-configmap__ConfigMap",
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: []*unstructured.Unstructured{
+				{
+					Object: map[string]interface{}{
+						"apiVersion": "apps/v1",
+						"kind":       "Deployment",
+						"metadata": map[string]interface{}{
+							"name":      "my-deployment",
+							"namespace": "my-namespace",
+						},
+					},
+				},
+				{
+					Object: map[string]interface{}{
+						"apiVersion": "v1",
+						"kind":       "ConfigMap",
+						"metadata": map[string]interface{}{
+							"name":      "my-configmap",
+							"namespace": "my-other-namespace",
+						},
+					},
+				},
+			},
+			expectedErr: nil,
+		},
+	}
+
+	for _, tc := range testCases {
```

Comment:
This could be simplified to just use `tt` and `t` as normal?

```go
for _, tt := range testCases {
    t.Run(tt.name, func(t *testing.T) {
      gg := NewGomegaWithT(t)
       // rest of code
    })
}
```